lib/cn/cn_tree.c

```diff
@@ -841,16 +842,34 @@ cn_tree_lookup(
             }
         }
 
-        if (cn_node_isleaf(node))
-            break;
+        if (cn_node_isleaf(node)) {
+            int rc;
+
+            if (!is_probe)
+                break;
+
+            rc = route_node_keycmp_prefix(kt->kt_data, kt->kt_len, rn);
+            if (rc < 0)
+                break;
+
+            assert(rc == 0);
 
-        node = cn_tree_node_lookup(tree, kt->kt_data, kt->kt_len);
+            rn = route_node_next(rn);
+            if (!rn)
+                break;
+
+            node = route_node_tnode(rn);
+
+        } else {
+
+            node = cn_tree_node_lookup(tree, kt->kt_data, kt->kt_len);
+        }
     }
 
   done:
     rmlock_runlock(lock);
 
-    if (qctx) {
+    if (is_probe) {
```

Comment:
I am think we should split this function into two, one for normal, and one for probe...

Reply:
Yes, makes sense.